docs/contributing.md

```diff
@@ -0,0 +1,121 @@
+# Development and Contributing
+
+
+## Development Environment
+The project builds on MacOS, Windows, and Linux. Please note that cross-platform builds work on all
+of these platforms, however, MacOS packages built on Windows and on Linux will not have a proper 
+icon, as generation of MacOS icons is a non-trivial task on these platforms.
+
+<br/>
+
+### Code Style Guide
+Please note that you should not use a code auto-formatter in this codebase. Please read the following
+documents to learn more about reasons behind this decision and the recommended code style guide. 
+Be sure to carefully read the documents before contributing to this repository:
+- [Rust style guide 1](https://github.com/luna/basegl/blob/master/docs/style-guide.md)
+- [Rust style guide 2](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) 
+
+<br/>
+
+### Setup
+- **The Rust Toolchain**  
+  This project uses several features available only in the nightly Rust toolchain.
+Please use the [the Rust toolchain installer](https://rustup.rs) to install it:
+
+  ```bash
+  rustup toolchain install nightly-2019-11-04 # Install the nightly channel.
+  rustup default nightly                      # Set it as the default one.
+  rustup component add clippy                 # Install the linter.
+  ```
+
+- **Node Version Manager**  
+  In order to build the web and desktop applications you will need [node](https://nodejs.org) and 
+[npm](https://www.npmjs.com). Please note that the versions available in your system package manager
+will probably crash while building native-extensions. The only known stable solution is to use the 
+[Node Version Manager](https://github.com/nvm-sh/nvm). Please note that installing it from any 
```

Comment:
TThe NVM explicitly states that it doesn't support Windows. Alternatives should be explored. Or consequences of not using it properly explained.